### DISCUSSION TITLE
Does the courier have a TTL for an email?

### DISCUSSION THREAD

**Discussion by pavlovmike** [ory/kratos#932] [Q&A] [answered]
Some users register with incorrect emails, so will the Kratos disable confirmation emails due to the inability to send? Or those emails will be forever garbage in sending queue and server logs?

**Comment by zepatrik** [comment, marked as answer]
Here you have the message struct, it does not store any TTL or retry count: https://github.com/ory/kratos/blob/master/courier/message.go#L22-L34
And here is the couriers strategy on a send error: https://github.com/ory/kratos/blob/master/courier/courier.go#L152-L161
So I guess the message will be send indefinitely unless you delete it from the database (possible workaround). We should probably implement some strategy to avoid that.

**Reply by pavlovmike** [reply]
Do you need me to create an issue about that - to plan this in a future release?

**Reply by aeneasr** [reply]
An issue is a great start! A PR would be even better :)
